### PR TITLE
Binary_Insertion_Sort.js

### LINKED ISSUE #2125: Binary Insertion Sort
- [x] C
- [x] C++
- [x] C# - @Yashank18
- [ ] CoffeeScript
- [ ] Dart - @raksha009
- [ ] Go - @KamalDGRT
- [ ] Java - @raktim00
- [ ] JS - @sheetalssr
- [ ] Kotlin
- [ ] Php - @goyalmuskan
- [x] Python
- [ ] Readme - @goyalmuskan
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Sheetal-ssr**
Fixes #2125
#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- add a file Binary_Insertion_Sort.js

#### Language Used:
- javascript

#### File Added:
- Binary_Insertion_Sort.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
please remove the file which is not relevant to this PR

**Comment by sheetalssr** [comment]
> please remove the file which is not relevant to this PR

DONE.

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+//Binary Insertion sort is a variant of Insertion sorting in which proper location to insert
+//the selected element is found using the binary search.
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: make this as a multi line commend and keep a line space after it

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+var i=0;
+//taking the size of the array dynamically
+var n=prompt("enter the size of your array");
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space around operators

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+var arr = [];
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep a line space after this

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+for(var i=0; i<n; i++)
+{
+    //Taking Input from user
+    arr[i] = prompt('Enter Element ' + (i+1));
+}
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space around operators

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+function binarysearch(array,item,start,end)
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space after comma

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+{
+    if(end <= start)
+        return (item > array[start])? (start+1): start;
+    var mid = (start+end)/2;
+    if(item == array[mid])
+        return mid+1;
+    if(item >array[mid])
+        return binarysearch(array,item,mid+1,end);
+    return binarysearch(array,item,start,mid-1);
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space around operator 
keep space after comma

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        key = arr[i];
+        //to find location where key sould be inserted
+        location = binarysearch(arr,key,0,j);
+         // to move all elements after location to create space for key
+        while(j >= location)
+        {
+            array[j+1] = array[j];
+            j--;
+        }
+        array[j+1] = key;
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space around operator
keep space after comma

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+insertionsort(arr,n);
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: keep space after comma

File: Binary_Insertion_Sort/Binary_Insertion_Sort.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+console.log(arr);
+
+//INPUT:Enter the size of the array 5
```
Comment (asha15) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.js]: Take  `Enter the size of the array ` sentence to the next line

Comment (asha15) [reply Binary_Insertion_Sort/Binary_Insertion_Sort.js]: make this as a multi line commend

**Comment by MastersAbh** [comment]
@Sheetal-ssr I dont see any update for 25 days hence, closing this for now. If you want to work on this, please reach out.

**Comment by Sheetal-ssr** [comment]
Sorry ,for the delay.I am on it now.i will be giving another pull request in a day or two .

**Pull request closed without merging**